lib/features/cli/networking/sync_session.dart

```diff
@@ -0,0 +1,23 @@
+import 'dart:async';
+
+import '../../../core/async/event_synchronizer.dart';
+import '../../../core/networking/websockets/session.dart';
+
+/// Creates an [EventSynchronizer] for a [ClientSession].
+///
+/// An optional [messageHandler] can be assigned which executes on
+/// every message received from the server.
+EventSynchronizer syncSession(
+  ClientSession session, [
+  FutureOr<void> Function(dynamic)? messageHandler,
+]) {
+  final sync = EventSynchronizer(session.send, Duration(milliseconds: 100));
+
+  // Pass all messages received by 'session' to the synchronizer
+  session.onMessage = (session, message) async {
+    await messageHandler?.call(message);
```

Comment:
You know that you can return data directly from the `onMessage` callback and don't need to `send` again? Or do you think this wouldn't make sense for the majority of CLI use cases?